### PR TITLE
tpm2-util: Only assume system support if we can dlopen() the tpm libraries

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

File: src/shared/tpm2-util.c [hunk 1]
```diff
@@ -4194,7 +4194,9 @@ Tpm2Support tpm2_support(void) {
                 support |= TPM2_SUPPORT_FIRMWARE;
 
 #if HAVE_TPM2
-        support |= TPM2_SUPPORT_SYSTEM;
+        r = dlopen_tpm2();
+        if (r >= 0)
+                support |= TPM2_SUPPORT_SYSTEM;
```
Comment (poettering) [inline src/shared/tpm2-util.c]: hmm, we could also add another flag for that. i.e. might be interesting to distinguish "systemd would support it if you installed some more .so" and "systemd would not support it, fuck you"...

i.e. add TPM2_SUPPORT_LIBRARIES or so

File: src/shared/tpm2-util.h [hunk 2]
```diff
@@ -183,7 +183,8 @@ typedef enum Tpm2Support {
         TPM2_SUPPORT_DRIVER    = 1 << 1,  /* the kernel has a driver loaded for it */
         TPM2_SUPPORT_SYSTEM    = 1 << 2,  /* we support it ourselves */
         TPM2_SUPPORT_SUBSYSTEM = 1 << 3,  /* the kernel has the tpm subsystem enabled */
-        TPM2_SUPPORT_FULL      = TPM2_SUPPORT_FIRMWARE|TPM2_SUPPORT_DRIVER|TPM2_SUPPORT_SYSTEM|TPM2_SUPPORT_SUBSYSTEM,
+        TPM2_SUPPORT_LIBRARIES = 1 << 4,  /* we can dlopen the tpm2 libraries */
+        TPM2_SUPPORT_FULL      = TPM2_SUPPORT_FIRMWARE|TPM2_SUPPORT_DRIVER|TPM2_SUPPORT_SYSTEM|TPM2_SUPPORT_SUBSYSTEM|TPM2_SUPPORT_LIBRARIES,
```
Comment (poettering) [inline src/shared/tpm2-util.h:187]: please update verb_has_tpm2() in src/creds/creds.c too

File: src/creds/creds.c [hunk 3]
```diff
@@ -636,11 +636,13 @@ static int verb_has_tpm2(int argc, char **argv, void *userdata) {
                 printf("%sfirmware\n"
                        "%sdriver\n"
                        "%ssystem\n"
-                       "%ssubsystem\n",
+                       "%ssubsystem\n"
+                       "%slibraries\n",
```
Comment (yuwata) [inline src/creds/creds.c:640]: The change to creds.c should be in the first commit?

Comment (daandemeyer) [reply src/creds/creds.c:640]: Ugh I keep forgetting to rebase instead of amend

**Pull request merged**